### PR TITLE
Fix repository path lookup when custom Hatchet directory set

### LINKED ISSUE #161: `path_for_name` doesn't take into account hatchet.json directory config
It's possible to override the default `REPOS_DIRECTORY_ROOT` of `"."` using:

```json
{
  "hatchet": {
    "directory": "custom/path"
  }
}
```

This is used by in most places, apart from `path_for_name`, which hardcodes `repos/` being in the project root:
https://github.com/heroku/hatchet/blob/97a18b873e9c0c5bcf3176a68b7b5ea79df77e5d/lib/hatchet/config.rb#L47

It seems this should instead be:

```ruby
      possible_paths = [repos[name.to_s], "#{repo_directory_path}/#{name}", name].compact
```

Or even better, it seems that the alternative style repo paths (presumably required for cases where there are multiple repos with the same short name) are added to `repos` itself by `init_config!`:
https://github.com/heroku/hatchet/blob/97a18b873e9c0c5bcf3176a68b7b5ea79df77e5d/lib/hatchet/config.rb#L71-L87

That way the affected line in `path_for_name` could be simplified to:

```ruby
      possible_paths = [repos[name.to_s], name].compact
```

...since the `repos` mapping would contain all of the variants itself.

### PR DESCRIPTION
**Pull request by edmorley**
Previously if a custom Hatchet `directory` was defined in `hatchet.json`, passing a fixture name of the full repository name would not work.

Fixes #161.

### CODE REVIEW

File: spec/hatchet/config_spec.rb [hunk 1]
```diff
@@ -6,6 +6,10 @@
     expect(@config.path_for_name("rails3_mri_193")).to(eq("repo_fixtures/repos/rails3/rails3_mri_193"))
   end
 
+  it("config path for name with full repo name") do
```
Comment (schneems) [inline spec/hatchet/config_spec.rb:9]: Style nit. The rspec methods don't use parens `it` `describe` etc. It reads fine, not enough to block merging. But it did stand out.

We could consider adding a style linter like standard-rb/prettier/rubocop. Setting them up is always a pain though. The smaller the team that modifies the source, the less value linters provide IMHO.

Comment (edmorley) [reply spec/hatchet/config_spec.rb:9]: Ah I missed there were parens. I'd copied/pasted the block above, which also has parens.

I have rubocop + the rspec rubocop plugin enabled for the Python repo. I found the onbaording experience excellent with Rubocop compared to other linters -- I used the auto-generated "todos" file + autocorrect to get most the way there, then permanently disabled any cops that were still failing and didn't seem to add value.

**Comment by schneems** [comment]
thanks!

File: spec/hatchet/config_spec.rb (hunk 1 above)
Comment (schneems) [reply spec/hatchet/config_spec.rb:9]: Ahh. Originally the tests were written in minitest. But the ecosystem around plugins (i.e. parallel testing) isn't as robust as rspec. I used a tool to auto-convert one test syntax to the other. I think it added the parens.

**Pull request merged**